plugins/org.jkiss.dbeaver.model/src/org/jkiss/dbeaver/utils/SystemVariablesResolver.java

```diff
@@ -41,6 +42,13 @@
     public static final String VAR_DBEAVER_HOME = "dbeaver_home";
     public static final String VAR_LOCAL_IP = "local.ip";
 
+    private Properties configurationFile = null;
+
+    public void setConfigurationFilePath(Properties configurationFile) {
+        this.configurationFile = configurationFile;
+        INSTANCE = this;
```

Comment:
What's the point of this assignment? The `INSTANCE` should be `final`, by the way.

Also, a private constructor would be nice to have to ensure the user will use the singleton.